test/integration/custom_targets.cpp

```diff
@@ -0,0 +1,85 @@
+#include "hsm/hsm.h"
+
+#include <boost/hana.hpp>
+#include <boost/hana/experimental/printable.hpp>
+#include <gtest/gtest.h>
+
+#include <future>
+#include <memory>
+
+namespace {
+
+using namespace ::testing;
+using namespace boost::hana;
+
+// States
+struct S1 {
+    std::string name = "S1";
+};
+struct S2 {
+    S2(std::string name)
+        : name(name)
+    {
+    }
+    std::string name;
```

Comment:
Can we make this member `const` to make the test even more demanding?

Reply:
- [ ] make const